### DISCUSSION TITLE
meow-(next|previous)-(word|symbol) without spaces

### DISCUSSION THREAD

**Discussion by icp1994** [meow-edit/meow#661] [Q&A] [answered]
I think this behavior was changed (fixed?) recently.
![image](https://github.com/user-attachments/assets/da975014-c207-4dc8-b5e9-a9a4307762e2)

Can this be customized to not include the spaces in-between?

Also meow-(next|previous)-word now goes over `.`
![image](https://github.com/user-attachments/assets/21ae693c-b875-4c88-8ac9-e16254751b93)

**Comment by DogLooksGood** [comment]
Yep, it's actually a fix.  By the original design, you can always use `mark-word/symbol` to shrink your selection.  It was affected when introducing a refactor or `meow-next-thing`. 

If want a command to just select the word, you can define a command like this.

```
(defun meow-next-word-1 (n)
  (interactive "p")
   ;; The optional last arg, default value is "w_", C-h s for info about syntax table
  (meow-next-thing meow-word-thing 'word n "w")) 
```

BTW, just use `mark-word/symbol`. The default set of word/symbol commands can cover all use cases.

**Comment by icp1994** [comment]
I quite often used it for editing the next/previous word/symbol, i.e, `e` `c` `<new-symbol>`. But now it is `e` `w` `c` `<new-symbol>`. It would be helpful if a var like `meow-include-syntax` can be be exposed for configuration.

**Reply by DogLooksGood** [reply]
I see no bad to add an option. BTW how you change/delete the word and the space next to it?

**Reply by icp1994** [reply]
`e` `c` `Backspace` if I want to append or `e` `s` `D` to just kill.

**Reply by DogLooksGood** [reply, marked as answer]
The variable `meow-next-thing-include-syntax` is added.

**Reply by icp1994** [reply]
Thank you <3
